superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterValue.tsx

```diff
@@ -137,7 +137,7 @@ const FilterValue: React.FC<FilterProps> = ({
         width={220}
         formData={formData}
         // For charts that don't have datasource we need workaround for empty placeholder
-        queriesData={hasDataSource ? state : [{ data: [null] }]}
+        queriesData={hasDataSource ? state : [{ data: [{}] }]}
```

Comment:
The type in `ChartDataResponseResult` is `data: DataRecord[]`, so better to return an empty row.